filesmanager/filesmanager.py

```diff
@@ -658,6 +743,34 @@ def get_formatted_content(self):
             "treeFolders": self.directories,
         }
 
+    def generate_asset_name(self, path):
+        return f"files-{self.block_id_parsed}-{path.replace('/', '-')}"
+
+    def is_in_filesmanager(self, course_asset):
+        """Check if an asset is part of the xblock content.
+
+        Arguments:
+            course_asset: the course asset to be checked.
+
+        Returns: True if the asset key is in the files manager, False otherwise.
+        """
+        if course_asset["asset_key"] in self.source_keys:
+            return True
+
+        display_name = course_asset["display_name"]
+        is_num = False
```

Comment:
what's this variable for?

Reply:
The XBlock ID is a UUID, UUIDs are hexadecimal numbers, so they can be cast to integers. This variable is to verify if what follows `files-` in the display name of the asset is a number. If it's it matches `files-{xblock_id}` and it's uploaded from files-manager and needs to be ignored from the Unpublished directory